app/registries/foreman/plugin/importer_registry.rb

```diff
@@ -0,0 +1,26 @@
+module Foreman
+  class Plugin
+    class ImporterRegistry
+      attr_accessor :importers 
+
+      def initialize
+        @importers = {}
+        @importers.default = 'PuppetFactImporter'
+      end
+
+      def register(key, klass, default = false)
+        @importers.default = klass if default
+        @importers[key.to_sym] = klass
+      end
+      alias register_importer register
+
+      def get(key)
+        @importers[key].constantize
+      end
+
+      def fact_features
+        importers.map { |_type, importer| importer.authorized_smart_proxy_features }.compact.flatten.uniq
```

Comment:
I think there is `constantize` missing here, `importers` has strings as values, not classes.

Reply:
@ShimShtein  so the idea here was that when i `get` the importer i constantize them. So ihm, there is not need to do constantise here.